packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -196,7 +195,7 @@ class OnsetPicker extends React.Component {
    * Update onset date when date changes
    */
   changeDate(event, date) {
-    const newDate = moment(date);
+    const newDate = moment(date).format('YYYY-MM-DD');
```

Comment:
Can we make this date format assumption?

Reply:
Hm well this is the onChange handler for the terra-datepicker, so unless that component is returning an ill-formatted date then it should be fine.

Reply:
@benbcai

Reply:
onChange in terra-datepicker either returns a valid ISO8601 string when a valid date is entered/selected or an empty string if the date is cleared. If the date is cleared, you would get an empty string and moment(date).format('YYYY-MM-DD') would return "Invalid date". You could check if newDate is valid by calling newDate.isValid(). If it is valid, then format it. Otherwise, set onsetDate to an empty string.

Reply:
Should the date format depend on locale?

Reply:
The format of the date returned from onChange in terra-date-picker is always in ISO8601 so that consumers can expect a consistent format.
It looks like the format ('YYYY-MM-DD') used here is to set the onsetDate state. The onsetDate state is only used internally for date manipulations and not displayed to the user.